api/controllers/game/rematch.js

```diff
@@ -22,14 +22,21 @@ module.exports = async function (req, res) {
     // Determine whether to start new game
     const oldPNum = game.p0.id === userId ? 0 : 1;
     const rematchVal = { [`p${oldPNum}Rematch`]: rematch };
-    const gameUpdates = {
-      ...rematchVal,
-      lastEvent: {
-        change: 'rematch',
-        game: { ...game.lastEvent.game, ...rematchVal },
-        victory: game.lastEvent.victory
-      }
-    };
+
+    let gameUpdates;
+
+    if (process.env.VITE_USE_GAMESTATE_API) {
+      gameUpdates = { ...rematchVal };
+    } else {
+      gameUpdates = {
+        ...rematchVal,
+        lastEvent: {
+          change: 'rematch',
+          game: { ...game.lastEvent.game, ...rematchVal },
+          victory: game.lastEvent.victory
+        }
+      };
+    }
```

Comment:
My instinct is to create a new version of this endpoint altogether and similarly transform it in `makeSocketRequest()` so that the environment-specific behavior is encapsulated in different endpoints rather than conditional logic in the same endpoint. 

Thoughts?